docs/Extension.md

```diff
@@ -0,0 +1,96 @@
+# Extension
+
+Sometimes it would be great to have ability to explore what payload is sent to ReportPortal server and update it or add extra one based on your own conditions. `ReportPortal.Shared` package provides you this ability by implementing `IReportEventsObserver` interface.
+
+## How to use own observers?
+
+To be able to observe events, next conditions should be met:
+1. Implement `IReportEventsObserver` interface
+2. Name of an assembly with implementation from previous step should contain `ReportPortal`
+3. The assembly should be in the same directory with `ReportPortal.Shared.dll`
+
+## Examples
+
+#### 1. Issue linking
+
+Let's imagine that you have implemented an `IssueAttribute` class which stores information about an issue in some Bug Tracking System. It would be great to automatically assign this issue to test in ReportPortal launch.
+
+Next code snippet shows how it can be done:
+
+```cs
+public class ReportPortalEventsObserver : IReportEventsObserver
+{
+    public void Initialize(IReportEventsSource reportEventsSource)
+    {
+        reportEventsSource.OnBeforeTestFinishing += ReportEventsSource_OnBeforeTestFinishing;
+    }
+
+    private void ReportEventsSource_OnBeforeTestFinishing(ITestReporter testReporter, BeforeTestFinishingEventArgs args)
+    {
+        var methodInfo = TestExecutionContext.CurrentContext.CurrentTest.Method;
```

Comment:
Example should be independent on unit test framework. BTW this example will not even work.

Reply:
Updated by replacing on more generic example.